Remove auction support routes II

dblandin opened the pull request:
Follow-up to https://github.com/artsy/force/pull/4921

We've recently moved registration and bidding functionality to [artsy/reaction](https://github.com/artsy/reaction). These routes and components are no longer used and should be cleaned up.

jira :lock:: [AUCT-493][]

[AUCT-493]: https://artsyproduct.atlassian.net/browse/AUCT-493

Todo:

- [x] Deploy to new review app
- [x] QA registration and bidding on mobile web
- [x] QA registration and bidding on desktop web

Registration QA should include:
- No cards on file
- Cards on file
- From artwork page
- From sale page

**Screen Recordings**

**desktop web / register / card on file**

![Peek 2020-01-15 15-58](https://user-images.githubusercontent.com/123595/72471869-73d06e00-37b1-11ea-874d-5fb9b2fb68ac.gif)

---

**desktop web / register + bid / card on file**

![Peek 2020-01-15 16-01](https://user-images.githubusercontent.com/123595/72471870-74690480-37b1-11ea-8d82-67bf5bff5f4d.gif)

---

**desktop web / register / no card on file**

![Peek 2020-01-15 16-07](https://user-images.githubusercontent.com/123595/72471877-7763f500-37b1-11ea-853e-447d750c3fb9.gif)

---

**desktop web / register + bid / no card on file**

![Peek 2020-01-15 16-09](https://user-images.githubusercontent.com/123595/72471878-7763f500-37b1-11ea-8346-e1a33b710ce5.gif)

---

**mobile web / register / card on file**

![Peek 2020-01-15 16-26](https://user-images.githubusercontent.com/123595/72473405-9d3ec900-37b4-11ea-99de-240e9beaf6ee.gif)

---

**mobile web / register / no card on file**

![Peek 2020-01-15 16-27](https://user-images.githubusercontent.com/123595/72473417-a29c1380-37b4-11ea-8327-fb2dba2d6dda.gif)

---

**mobile web / register + bid / card on file**

![Peek 2020-01-15 16-28](https://user-images.githubusercontent.com/123595/72473423-a7f95e00-37b4-11ea-9e12-b339086a4311.gif)

---

**mobile web / register + bid / no card on file**

![Peek 2020-01-15 16-31](https://user-images.githubusercontent.com/123595/72473427-ab8ce500-37b4-11ea-8974-670d37d961e9.gif)

erikdstock commented on src/mobile/apps/feature/routes.coffee:
```diff
@@ -1,133 +1,9 @@
-    error: res.backboneError
-
-module.exports.auctionRegister = (req, res, next) ->
-  # If not logged in, send them to sign up to return to this handler
```
did we figure out where the `feature` app was used?

dblandin replied:
I've found a few recent examples:

https://www.artsy.net/feature/discover-art-from-stuttgart
https://www.artsy.net/feature/laguna-art-museum-california-cool-benefit-auction-2020
https://www.artsy.net/feature/whitechapel-gallery-art-icon-benefit-auction-2020

From what I can tell, no features currently support registration or bidding directly on the feature page. Auctions have a corresponding `Feature` but the feature page `/feature/:feature_slug` now redirects you to the `auction/:auction_slug` route.

dblandin commented on src/desktop/apps/auction_support/routes.coffee:
```diff
@@ -35,37 +29,24 @@ registerOrRender = (sale, req, res, next) ->
 
 
-@auctionRegistration = (req, res, next) ->
+@modalAuctionRegistration = (req, res, next) ->
```
@erikdstock This was removed in the diff in the last PR which broke registration on the sale page when a user already has a credit card on file. This route is currently triggered via the modal on the sale page, so I've renamed this route handler and kept the surrounding code (thankfully not much) in.

dblandin commented:
> I realize this is still pending due to merge conflicts but tentatively approve

I addressed those merge conflicts. Looks like https://github.com/artsy/force/pull/4933 updated a few files I removed or updated.

zephraph commented:
Doing the Lord’s work here 🙌

🦾

dleve123 commented on src/desktop/analytics/artwork.js:
```diff
@@ -522,9 +522,4 @@ import { data as sd } from "sharify"
     .on("artwork:contact-specialist", function() {
       analytics.track("Clicked 'Contact Artsy Specialist'")
     })
-    .on("artwork:confirm-registration", function() {
```
Do we know if there's any corresponding instrumentation in Reaction now? Was the only call-sight to this hook from code that was deleted in this PR? Just wanna make sure we're not deleting an analytics event

dblandin replied:
Yes, we have page tracking in reaction. I don't see any explicit examples of this event triggered in either Force or Reaction. Might have been fired dynamically using the route and model passed in. Not sure but either way should be safe to remove.

dleve123 replied:
Yeah, I guess it wasn't be used for a while anyways in Force, even if the code was there.

dblandin commented:
Merging this in :tada:
